capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -54,21 +56,27 @@ public static JsonObject createRequestBodyJson(HttpServletRequest request) throw
 
   /**
    * Query the given collection by id. 
+   * @param {id} id to be searched 
    * @param {collectionReference} reference to the appropriate database collection
    * @param {request} request sent to the backend
    * @param {response} response returned from the call
    * @param {genericClass} a Generic class, used in casting documents from the database
    * @return List<T> a singleton List of the Object that matches the ID in the request.
    */
-  private static <T extends BaseEntity> List<T> getById(CollectionReference collectionReference, HttpServletRequest request, 
-      HttpServletResponse response, GenericClass<T> genericClass) throws IOException {
+  private static <T extends BaseEntity> List<T> getById(String id, CollectionReference collectionReference, 
+      HttpServletRequest request, HttpServletResponse response, GenericClass<T> genericClass) throws IOException {
+    if (id.length() == 0) {
+      System.err.println("Error caused by either an empty or non-existent \"id\" field in the post body.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return null;
+    }
     if (request.getParameterMap().size() > 1) {
       System.err.println("Error: No other parameter can be sent with an ID");
       response.sendError(HttpServletResponse.SC_BAD_REQUEST);
       return null;
     }
```

Comment:
If you're now passing the id directly to the function, is it necessary to even pass the request as a parameter and to get its parameterMap? I'm guessing that you probably wouldn't be sending the id as a parameter and also as a field in the request.

Reply:
I could move the parameterMap check up into the get function, rather than leaving it here.